File: apis/src/main/java/com/beat/apis/performance/facade/HomeFacade.java
package com.beat.apis.performance.facade;

import org.springframework.stereotype.Service;

import com.beat.apis.performance.application.HomeService;
import com.beat.apis.performance.application.dto.home.HomeFindAllResponse;
import com.beat.apis.performance.application.dto.home.HomeFindRequest;
import com.beat.apis.performance.application.dto.GenreType;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class HomeFacade {
	private final HomeService homeService;

	public HomeFindAllResponse findHomePerformanceList(GenreType genre) {
		return homeService.findHomePerformanceList(new HomeFindRequest(genre));
	}
}

File: apps/api/src/main/kotlin/com/beat/apps/api/home/facade/HomeFacade.kt
package com.beat.apps.api.home.facade

import com.beat.application.frontoffice.home.booker.query.HomeQueryService
import com.beat.apps.api.home.api.response.HomeFindAllResponse
import com.beat.apps.api.home.api.type.HomeGenreType
import org.springframework.stereotype.Service

@Service
class HomeFacade(private val homeQueryService: HomeQueryService) {
    fun findHomePerformanceList(genre: HomeGenreType?): HomeFindAllResponse =
        HomeFindAllResponse.from(homeQueryService.findHomePerformanceList(genre?.name))
}
